### PR TITLE
feat: submit project integration with backend

### LINKED ISSUE #175: [FEATURE] integrate add-project page with backend api
### Description

Requirement: 

Integrate add-project form with the backend create project API. 
Use react-hook-form (https://react-hook-form.com/)
Add validation in frontend. 
for now use any hardcoded blank image url while adding project.

### PR DESCRIPTION
**Pull request by rohitdasu**
### Fixes Issue #175 
closes #175 

### Changes proposed

1. Project Page structure improved
2. Removed toast message - due to some issues with that
3. re-written add project page
4. added react-hook-form
5. added validation
6. integrated with add project api

### CODE REVIEW

**Review by leewan-09 — COMMENTED** [review]
Also, this route should be protected.  if a user is not logged in then redirect to the home page.
you can achieve with this:
https://next-auth.js.org/getting-started/client#example-1

File: pages/projects/submit.tsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    reset,
+    formState: { errors },
+  } = useForm<FormInputs>();
+  const { data: session } = useSession();
+
+  const [tagInput, setTagInput] = useState<string>('');
+  const removeTag = (index: number) => {
+    setValue(
+      'tags',
+      watch('tags').filter((element, i) => i !== index)
```
Comment (leewan-09) [inline pages/projects/submit.tsx:33]: No types?

**Comment by rohitdasu** [comment]
> Also, this route should be protected. if a user is not logged in then redirect to the home page. you can achieve with this: https://next-auth.js.org/getting-started/client#example-1

good point

File: pages/projects/submit.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    );
+  };
+
+  const onSubmit: SubmitHandler<FormInputs> = (data) => {
```
[the remark this answers is not in the document]
Comment (rohitdasu) [reply pages/projects/submit.tsx:37]: sure

[the remark this answers is not in the document]
Comment (leewan-09) [reply pages/projects/submit.tsx:37]: Give type to input `(data:FormInputs)`

File: pages/projects/submit.tsx (hunk 1 above)
Comment (rohitdasu) [reply pages/projects/submit.tsx:33]: ah, will do it 👍

**Comment by leewan-09** [comment]
Also i think image have less padding 
image

**Comment by rohitdasu** [comment]
> Also i think image have less padding image

you mean margin?

**Review by leewan-09 — COMMENTED** [review]
* I see this add project page 
image

* Should it be active on the add project page?
image

* I think error messages are too large 
image

* No active border on the input field. But you see an active border on the description
image

image

File: pages/projects/submit.tsx [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    control,
+    reset,
+    formState: { errors },
+  } = useForm<FormInputs>();
+  const { data: session } = useSession();
```
[the remark this answers is not in the document]
Comment (rohitdasu) [reply pages/projects/submit.tsx:28]: sure

**Review by leewan-09 — APPROVED** [review]
LGTM 👍

**Pull request merged**